### ISSUE TITLE
Feature: Unsafe version of `imencode()` that does not copy

### ISSUE THREAD

**Issue by Levi-Lesches** [rainyl/opencv_dart#304]
**Is your feature request related to a problem? Please describe.**
The [`imencode()`](https://pub.dev/documentation/opencv_dart/latest/cv/imencode.html) function has a line like this: 

https://github.com/rainyl/opencv_dart/blob/4983c38199402dfbe74572b05966489121a8db6c/packages/dartcv/lib/src/imgcodecs/imgcodecs.dart#L107-L110

I'm writing [a program](https://github.com/BinghamtonRover/Video) that tries to read from 8+ cameras, on a Raspberry Pi 5, at ~30 FPS. My old implementation passed a `(Pointer<Uint8> data, int length)` directly between isolates to avoid copying, but now I'm afraid this will introduce latency from copying all those frames.

Quick math: (8 cameras) * (30 frames / second / camera) * (60,000 bytes / frame) = 14 Mb / sec.

**Describe the solution you'd like**
A function like the following could return the pointer itself, allowing me to pass just a single integer across isolates and send the data from native memory, avoiding all copies entirely, so long as I dispose of everything properly. 

```dart
(bool, VecUChar) imencodeUnsafe(...);
```

**Describe alternatives you've considered**
On a Windows laptop with just one integrated webcam, the current code works fine. I have yet to test this on on my 8+ cameras on a Raspberry Pi 5, so I imagine that copying will introduce unacceptable delays. I am happy to open a PR if you'd like. 

It might be worth considering a breaking change in the future to make this the default behavior. Similar to `VideoCapture.read()` which returns a `Mat`, I think it's okay to let the caller dispose of their data manually, especially if it means avoiding some major copying.

**Additional context**

This might help my use case even more: I recently discovered [`@pragma('vm:deeply-immutable')`](https://github.com/dart-lang/sdk/blob/main/runtime/docs/deeply_immutable.md), which allows some objects to not be copied between isolates, but shared. As far as I know, a `Uint8List` does not qualify, but if I use `vec.ptr.address`, that will work, and I can eliminate _all_ copying across isolates (9+ isolates) entirely!

**Comment by rainyl** [comment]
Yes it's applicable, I just didn't take it into consideration when designing, that's what I said `can be improved`. 😢 

However I have no much time to reconstruct the vector related wrappers recently, it would be nice if you can work on this. 😄

**Comment by Levi-Lesches** [comment]
Sounds good, I'll submit a PR soon

**Comment by Levi-Lesches** [comment]
Oh, thanks for doing #305 for us! So all that's left for me to do is change all the `vecXXX_cpp2c` to be non-copying, right?

**Comment by rainyl** [comment]
> Oh, thanks for doing #305 for us! So all that's left for me to do is change all the `vecXXX_cpp2c` to be non-copying, right?

Exactly. But it should be done in dartcv, so please submit PR there, we can keep tracking at https://github.com/rainyl/dartcv/issues/14
